optimum/habana/diffusers/pipeline_utils.py

```diff
@@ -0,0 +1,270 @@
+
+
+GAUDI_LOADABLE_CLASSES = {
+    "diffusers": {
+        "ModelMixin": ["save_pretrained", "from_pretrained"],
+        "SchedulerMixin": ["save_pretrained", "from_pretrained"],
+        "DiffusionPipeline": ["save_pretrained", "from_pretrained"],
+        "OnnxRuntimeModel": ["save_pretrained", "from_pretrained"],
+    },
+    "transformers": {
+        "PreTrainedTokenizer": ["save_pretrained", "from_pretrained"],
+        "PreTrainedTokenizerFast": ["save_pretrained", "from_pretrained"],
+        "PreTrainedModel": ["save_pretrained", "from_pretrained"],
+        "FeatureExtractionMixin": ["save_pretrained", "from_pretrained"],
+        "ProcessorMixin": ["save_pretrained", "from_pretrained"],
+        "ImageProcessingMixin": ["save_pretrained", "from_pretrained"],
+    },
+    "optimum.habana.diffusers.schedulers": {
+        "GaudiDDIMScheduler": ["save_pretrained", "from_pretrained"],
+    },
+}
+
+GAUDI_ALL_IMPORTABLE_CLASSES = {}
+for library in GAUDI_LOADABLE_CLASSES:
+    GAUDI_ALL_IMPORTABLE_CLASSES.update(GAUDI_LOADABLE_CLASSES[library])
+
+
+class GaudiDiffusionPipeline(DiffusionPipeline):
+    """
+    Extends the [`DiffusionPipeline`](https://huggingface.co/docs/diffusers/api/diffusion_pipeline) class:
+    - The pipeline is initialized on Gaudi if `use_habana=True`.
+    - The pipeline's Gaudi configuration is saved and pushed to the hub.
+
+    Args:
+        use_habana (bool, defaults to `False`):
+            Whether to use Gaudi (`True`) or CPU (`False`).
+        use_lazy_mode (bool, defaults to `False`):
+            Whether to use lazy (`True`) or eager (`False`) mode.
+        use_hpu_graphs (bool, defaults to `False`):
+            Whether to use HPU graphs or not.
+        gaudi_config (Union[str, [`GaudiConfig`]], defaults to `None`):
+            Gaudi configuration to use. Can be a string to download it from the Hub.
+            Or a previously initialized config can be passed.
+    """
+
+    def __init__(
+        self,
+        use_habana: bool = False,
+        use_lazy_mode: bool = False,
+        use_hpu_graphs: bool = False,
+        gaudi_config: Union[str, GaudiConfig] = None,
+    ):
+        super().__init__()
+
+        # Lazy mode, HPU graphs and Gaudi configuration should be off if not using HPU
+        self.use_habana = use_habana
+        self.use_lazy_mode = False if not use_habana else use_lazy_mode
+        self.use_hpu_graphs = False if not use_habana else use_hpu_graphs
+        self.gaudi_config = None if not use_habana else gaudi_config
+
+        if self.use_lazy_mode and self.use_hpu_graphs:
+            raise ValueError("`use_lazy_mode` and `use_hpu_graphs` cannot be both True.")
+
+        if self.use_habana:
+            if self.use_lazy_mode:
+                logger.info("Enabled lazy mode.")
+            elif self.use_hpu_graphs:
+                logger.info("Enabled HPU graphs.")
+            else:
+                os.environ["PT_HPU_LAZY_MODE"] = "2"
```

Comment:
remove lazy_mode as 2 since it's eager mode

Reply:
Ah I may have misunderstood during the meeting yesterday: we want to remove eager mode and set lazy mode as the default right?